### PR TITLE
(dev/core#5365) Extension Upgrade - Pass-through status-result for custom upgraders

### PR DESCRIPTION
**Pull request by ufundo**
Fix for https://lab.civicrm.org/dev/core/-/issues/5365

### CODE REVIEW

**Comment by ufundo** [comment]
Maybe _you_ should check again, civibot... https://lab.civicrm.org/dev/core/-/issues/5365

**Comment by mlutfy** [comment]
jenkins, test this please

**Comment by ufundo** [comment]
Jenkins test again please

**Comment by ufundo** [comment]
Jenkins test this please

**Comment by totten** [comment]
civibot, test this please

**Comment by totten** [comment]
I believe the test failure is correct -- if you installed 5.39.1 with `afform/core/` and then upgraded to the current version, it wouldn't work. The bug was masked by the presence of the bug -- and unmasked by the fix. Pushed an update which should fix it...

File: ext/afform/core/sql/upgrade_1001.sql [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+-- *
+-- * Clean up the existing tables - this section generated from drop.tpl
+-- *
+-- *******************************************************/
+
+SET FOREIGN_KEY_CHECKS=0;
+
+DROP TABLE IF EXISTS `civicrm_afform_submission`;
+
+SET FOREIGN_KEY_CHECKS=1;
```
Comment (colemanw) [inline ext/afform/core/sql/upgrade_1001.sql:22]: I believe these lines are redundant with the conditional in `function upgrade_1001()` and as a general rule I think we should avoid these "loaded gun" self-destruct statements.

Comment (totten) [reply ext/afform/core/sql/upgrade_1001.sql]: @colemanw Yeah, that bit was just a blind restoration. But I think you're right. Updated.

**Comment by colemanw** [comment]
@totten I've given this merge-ready. I believe it should be backported to 5.80?

**Comment by totten** [comment]
Tested the same patch on 5.80, and it seemed to fix the issue there too.

I want to test one more angle and then merge.

**Comment by totten** [comment]
Tested a backport of this patch for civix (https://github.com/totten/civix/pull/379). There was  a small issue that would affect older extensions. (*In the interim between when this PR was written and now, the same subproject had an intervening change which affected php73*). Rebased and threw in a quick fix. 

When the set of #30740 + #31607 + https://github.com/totten/civix/pull/379 are passing, then we should be good to merge the set.

**Pull request merged**